src/tests.rs

```diff
@@ -158,118 +158,153 @@ mod platform {
+            }
 
-        if SetStdHandle(STD_ERROR_HANDLE, stdout) == 0 {
-            return Err(io::Error::last_os_error());
-        }
+            if SetStdHandle(STD_ERROR_HANDLE, stdout) == 0 {
+                return Err(io::Error::last_os_error());
+            }
 
-        OLD_OUT = Box::into_raw(Box::new(old_out));
+            OLD_OUT = Box::into_raw(Box::new(old_out));
 
-        Ok(())
+            Ok(())
+        }
     }
 
     /// Reattach to the old console.
-    pub unsafe fn cleanup() -> io::Result<()> {
-        if FreeConsole() == 0 {
-            return Err(io::Error::last_os_error());
+    pub fn cleanup() -> io::Result<()> {
+        unsafe {
+            if FreeConsole() == 0 {
+                return Err(io::Error::last_os_error());
+            }
+
+            if AttachConsole(winapi::um::wincon::ATTACH_PARENT_PROCESS) == 0 {
+                return Err(io::Error::last_os_error());
+            }
+
+            Box::from_raw(OLD_OUT).set_as_std()?;
+
+            Ok(())
+        }
+    }
+
+    /// This will signal the whole process group.
+    pub fn raise_ctrl_c() {
+        unsafe {
+            assert!(GenerateConsoleCtrlEvent(winapi::um::wincon::CTRL_C_EVENT, 0) != 0);
         }
+    }
 
-        if AttachConsole(winapi::um::wincon::ATTACH_PARENT_PROCESS) == 0 {
-            return Err(io::Error::last_os_error());
+    /// Print to both consoles, this is not thread safe.
+    pub fn print(fmt: ::std::fmt::Arguments) {
+        unsafe {
+            use self::io::Write;
+            {
+                let stdout = io::stdout();
+                stdout.lock().write_fmt(fmt).unwrap();
+            }
+            {
+                assert!(!OLD_OUT.is_null());
+                (*OLD_OUT).write_fmt(fmt).unwrap();
+            }
         }
+    }
+}
 
-        Box::from_raw(OLD_OUT).set_as_std()?;
+#[cfg(not(unix))]
+#[cfg(not(windows))]
+mod platform {
+    use std::io;
 
+    pub fn setup() -> io::Result<()> {
         Ok(())
     }
 
-    /// This will signal the whole process group.
-    pub unsafe fn raise_ctrl_c() {
-        assert!(GenerateConsoleCtrlEvent(winapi::um::wincon::CTRL_C_EVENT, 0) != 0);
+    pub fn cleanup() -> io::Result<()> {
+        Ok(())
     }
 
-    /// Print to both consoles, this is not thread safe.
-    pub unsafe fn print(fmt: ::std::fmt::Arguments) {
+    pub fn raise_ctrl_c() {
+    }
+
+    pub fn print(fmt: ::std::fmt::Arguments) {
         use self::io::Write;
-        {
-            let stdout = io::stdout();
-            stdout.lock().write_fmt(fmt).unwrap();
-        }
-        {
-            assert!(!OLD_OUT.is_null());
-            (*OLD_OUT).write_fmt(fmt).unwrap();
-        }
+        let stdout = ::std::io::stdout();
+        stdout.lock().write_fmt(fmt).unwrap();
     }
 }
 
-fn test_set_handler() {
-    let (tx, rx) = ::std::sync::mpsc::channel();
-    ctrlc::set_handler(move || {
-        tx.send(true).unwrap();
+async fn test_set_handler()
```

Comment:
fair enough - the original tests are unable to run more than one but i modified them so that they could in the last commit. this now means i can split the tests for sync and async